### PR TITLE
feat: support manifest and blob deletion for OCI layout

### LINKED ISSUE #761: support OCI image layout target in `oras blob delete`
User should be able to use `oras blob delete` to remove a blob in their OCI layout folder with properly garbage collection.

Depending on sdk support: https://github.com/oras-project/oras-go/issues/454

### LINKED ISSUE #762: support OCI image layout target in `oras manifest delete`
User should be able to use `oras manifest delete` to remove a manifest in their OCI layout folder with properly garbage collection.

### PR DESCRIPTION
**Pull request by qweeah**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #761, fixes #762 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/internal/option/target.go [hunk 1]
```diff
@@ -114,26 +116,69 @@ func parseOCILayoutReference(raw string) (path string, ref string, err error) {
...
-		var err error
-		opts.Path, opts.Reference, err = parseOCILayoutReference(opts.RawReference)
+		return opts.newOCIStore()
+	case TargetTypeRemote:
+		return opts.newRepository(common, logger)
+	}
+	return nil, fmt.Errorf("unknown target type: %q", opts.Type)
+}
+
+type resolvableDeleter interface {
```
Comment (shizhMSFT) [inline cmd/oras/internal/option/target.go:151]: Should we export this?

**Review by shizhMSFT — APPROVED** [review]
LGTM with suggestions

File: cmd/oras/internal/option/target.go (hunk 1 above)
Comment (qweeah) [reply cmd/oras/internal/option/target.go:151]: I think exporting `NewBlobDeleter` and `NewManifestDeleter` are already enough.

Comment (qweeah) [reply cmd/oras/internal/option/target.go:151]: We can add it to oras-go BTW, it's common to resolve before deletion and should be useful to other SDK callers.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**